barretenberg/cpp/src/barretenberg/api/bbapi_ultra_honk.hpp

```diff
@@ -0,0 +1,243 @@
+ * @brief Verify that a witness satisfies a circuit's constraints.
+ * For debugging and validation purposes.
+ */
+struct CircuitCheck {
+    static constexpr const char* NAME = "CircuitCheck";
+
+    struct Response {
+        static constexpr const char* NAME = "CircuitCheckResponse";
+
+        bool satisfied;
+    };
+
+    CircuitInput circuit;
+    std::vector<uint8_t> witness;
+    ProofSystemSettings settings;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @struct CircuitVerify
+ * @brief Verify a proof against a verification key and public inputs.
+ */
+struct CircuitVerify {
+    static constexpr const char* NAME = "CircuitVerify";
+
+    struct Response {
+        static constexpr const char* NAME = "CircuitVerifyResponse";
+
+        bool verified;
+    };
+
+    std::vector<uint8_t> verification_key;
+    PublicInputsVector public_inputs;
+    HonkProof proof;
+    ProofSystemSettings settings;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @struct ProofAsFields
+ * @brief Convert a proof to field elements representation.
+ */
+struct ProofAsFields {
+    static constexpr const char* NAME = "ProofAsFields";
+
+    struct Response {
+        static constexpr const char* NAME = "ProofAsFieldsResponse";
+
+        std::vector<bb::fr> fields;
+    };
+
+    HonkProof proof;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @struct VkAsFields
+ * @brief Convert a verification key to field elements representation.
+ */
+struct VkAsFields {
+    static constexpr const char* NAME = "VkAsFields";
+
+    struct Response {
+        static constexpr const char* NAME = "VkAsFieldsResponse";
+
+        std::vector<bb::fr> fields;
+    };
+
+    std::vector<uint8_t> verification_key;
+    bool is_mega_honk = false;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @brief Command to generate Solidity verifier contract
+ */
+struct CircuitWriteSolidityVerifier {
+    static constexpr const char* NAME = "CircuitWriteSolidityVerifier";
+
+    struct Response {
+        static constexpr const char* NAME = "CircuitWriteSolidityVerifierResponse";
+
+        std::string solidity_code;
+    };
+
+    std::vector<uint8_t> verification_key;
+    ProofSystemSettings settings;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @brief Command to prove and verify in one step
+ */
+struct CircuitProveAndVerify {
+    static constexpr const char* NAME = "CircuitProveAndVerify";
+
+    struct Response {
+        static constexpr const char* NAME = "CircuitProveAndVerifyResponse";
+
+        bool verified;
+        HonkProof proof;
+        PublicInputsVector public_inputs;
+    };
+
+    CircuitInput circuit;
+    std::vector<uint8_t> witness;
+    ProofSystemSettings settings;
+    Response execute(const BBApiRequest& request = {}) &&;
+};
+
+/**
+ * @brief Command to validate circuit structure
+ */
+struct CircuitValidate {
```

Comment:
AI slop perhaps?

Reply:
yeah im not really focusing on ultrahonk yet